src/directives/infinite-scroll/infinite-scroll.component.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { InfiniteScrollModule } from './infinite-scroll.module';
+import { Component } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+
+@Component({
+    template: `<div [uxInfiniteScroll]="load"
+                    [filter]="filterText"
+                    [pageSize]="20"
+                    [loadOnScroll]="false">
+                </div>
+    `
+})
+export class InfiniteScrollTestComponent {
+
+    filterText: any;
+
+    load(pageNum: number, pageSize: number, filter: any): any[] {
+        return [];
+    }
+
+}
+
+describe('Directive - Infinite Scroll', () => {
+    let component: InfiniteScrollTestComponent;
+    let fixture: ComponentFixture<InfiniteScrollTestComponent>;
+    let loadSpy: jasmine.Spy;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [InfiniteScrollModule, FormsModule],
+            declarations: [InfiniteScrollTestComponent],
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(InfiniteScrollTestComponent);
+        component = fixture.componentInstance;
+        loadSpy = spyOn(component, 'load').and.callThrough();
+        fixture.detectChanges();
+    });
+
+    it ('should initially call load with filter value of "" if filter input value is undefined', async() => {
+
+        expect(loadSpy).toHaveBeenCalledWith(0, 20, '');
```

Comment:
I wonder would it also be good to add some checks for the number of times the load function is called to ensure we don't try to load pages multiple times, eg:

```typescript
expect(loadSpy).toHaveBeenCalledTimes(1);
```